daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala

```diff
@@ -763,14 +763,17 @@ private[validation] object Typing {
       }
     }
 
-    private def typeOfLet(binding: Binding, body: Expr): Type = binding match {
-      case Binding(Some(vName), typ0, expr) =>
-        checkType(typ0, KStar)
-        val typ1 = checkExpr(expr, typ0)
-        introExprVar(vName, typ1).typeOf(body)
-      case Binding(_, _, bound @ _) =>
-        typeOf(body)
-    }
+    private def typeOfLet(bindings: List[Binding], body: Expr): Type =
+      bindings match {
+        case Binding(Some(vName), typ0, expr) :: tail =>
+          checkType(typ0, KStar)
+          val typ1 = checkExpr(expr, typ0)
+          introExprVar(vName, typ1).typeOfLet(tail, body)
```

Comment:
this doesn’t look stack safe, should we make it stack safe? I don’t expect the list of bindings to grow particularly large but it adds to an already large stack.

Reply:
We want, but another time